MaterialLib/SolidModels/CreateLubby2.h

```diff
@@ -93,20 +94,79 @@ std::unique_ptr<Lubby2<DisplacementDim>> createLubby2(
-        new Lubby2<DisplacementDim>{nonlinear_solver_parameters, mp}};
+    auto const dependency_parameter_mkt_name = config.getConfigParameterOptional<std::string>("dependency_parameter_mkt");
+
+    auto const dependency_parameter_q_name = config.getConfigParameterOptional<std::string>("dependency_parameter_q");
+
+    if(dependency_parameter_tref_name && dependency_parameter_mgt_name && dependency_parameter_mkt_name && dependency_parameter_q_name)
+    {
+        // Dependency parameter for Tref
+        auto& dependency_parameter_Tref = ParameterLib::findParameter<double>(*dependency_parameter_tref_name, parameters, 1);
+
+        DBUG("Use '%s' as dependency parameter Tref.",
+         dependency_parameter_Tref.name.c_str());
+
+        // Dependency parameter for mGT
+        auto& dependency_parameter_mGT = ParameterLib::findParameter<double>(*dependency_parameter_mgt_name, parameters, 1);
+
+        DBUG("Use '%s' as dependency parameter mGT.",
+            dependency_parameter_mGT.name.c_str());
+
+        // Dependency parameter for mKT
+        auto& dependency_parameter_mKT = ParameterLib::findParameter<double>(*dependency_parameter_mkt_name, parameters, 1);
+
+        DBUG("Use '%s' as dependency parameter mKT.",
+            dependency_parameter_mKT.name.c_str());
+
+        // Dependency parameter for Q
+        auto& dependency_parameter_Q = ParameterLib::findParameter<double>(*dependency_parameter_q_name, parameters, 1);
+
+        DBUG("Use '%s' as dependency parameter Q.",
+            dependency_parameter_Q.name.c_str());
+
+        Lubby2MaterialProperties mp{
+            kelvin_shear_modulus,     maxwell_shear_modulus,
+            maxwell_bulk_modulus,     kelvin_viscosity,
+            maxwell_viscosity,        dependency_parameter_mK,
+            dependency_parameter_mvK, dependency_parameter_mvM,
+            dependency_parameter_Tref, dependency_parameter_mGT,
+            dependency_parameter_mKT, dependency_parameter_Q};
+
+        auto const& nonlinear_solver_config =
+            //! \ogs_file_param{material__solid__constitutive_relation__Lubby2__nonlinear_solver}
+            config.getConfigSubtree("nonlinear_solver");
+        auto const nonlinear_solver_parameters =
+            createNewtonRaphsonSolverParameters(nonlinear_solver_config);
+
+        return std::unique_ptr<Lubby2<DisplacementDim>>{
+            new Lubby2<DisplacementDim>{nonlinear_solver_parameters, mp}};
+
+    }
+    else
+    {
+       auto parameter_empty_ptr = new ParameterLib::ConstantParameter<double>("NoName", 0.0);
+
+        Lubby2MaterialProperties mp{
+            kelvin_shear_modulus,     maxwell_shear_modulus,
+            maxwell_bulk_modulus,     kelvin_viscosity,
+            maxwell_viscosity,        dependency_parameter_mK,
+            dependency_parameter_mvK, dependency_parameter_mvM,
+            *parameter_empty_ptr, *parameter_empty_ptr,
+            *parameter_empty_ptr, *parameter_empty_ptr};
```

Comment:
I'd recommend to put the additional temperature related coefficients into a structure and pass them using `boost::optional`.
In the newly written code the use of plain pointers is prohibited, especially those causing memory leaks (like the above `parameter_empty_ptr`).

Reply:
or derive a class from Lubby2.